### PR TITLE
Add a short documentation section about the io sub-package

### LINKED ISSUE #53: io.h5 documentation apparently missing
Documentation about the io.h5 is apparently missing (at least, I could not find it on the online docs)

### PR DESCRIPTION
**Pull request by aaronayres35**
fixes #53 

This PR adds a _brief_ section to the documentation discussing the `apptools.io` and `apptools.io.h5` sub packages.
   
**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

File: docs/source/io/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+for common file path manipulation operations.  On Python 3 much of this
+functionality is provided by the `pathlib`_ module, and for new code we
+encourage users to investigate if `pathlib`_ can satisfy their usecase before
+they turn to the `apptools.io` |File| object
+
+io.h5
+-----
+The io.h5 sub-package provides a wrapper around `PyTables`_ with a
+dictionary-style mapping.  Note that this module has a decent amount of
+overlap with `zarr`_.
```
Comment (aaronayres35) [inline docs/source/io/introduction.rst:14]: We probably want to add more specifics here, but I am not exactly confident in what `io.h5` uniquely provides...

**Review by kitchoi — COMMENTED** [review]
Thank you. Some nitpicks, the reference to zarr might be an overstatement though.

File: docs/source/io/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
+io
```
Comment (kitchoi) [inline docs/source/io/introduction.rst:1]: Sorry I did not try to build the documentation earlier.
The all-small-cap and short section title looks slightly out of place:
Screenshot 2020-11-25 at 14 39 55

Would "File I/O" look a bit better? And "io.h5" be changed to "HDF5 File support"?

**Review by kitchoi — APPROVED** [review]
LGTM

**Pull request merged**